docs/lang/articles/type_system/type.md

```diff
@@ -158,11 +173,43 @@ When you assign a value to a variable of a different data type, Taichi implicitl
 
 ## Compound types
 
-Compound types are user-defined data types, which comprise multiple members. Supported compound types include vectors, metrics, and structs.
+Compound types are user-defined data types, which comprise multiple members. Supported compound types include vectors, matrices, ndarrays, and structs.
+
+Taichi allows you to use all types supplied in the `ti.types` module as scaffolds to customize *higher-level* compound types.
+
+:::note
+
+The `ndarray` type is discussed in another article [interacting with External Arrays](../basic/external.md).
 
-Taichi allows you to use all types supplied in the `ti.type` module as scaffolds to customize *higher-level* compound types.
+:::
+
+
+### Matrices and vectors
 
-Suppose you are using Taichi to represent a sphere. A sphere in the 3D space can be abstracted with its center and radius. In the following example, you call `ti.types.vector()` and `ti.types.struct()` to create compound types `vec3` and `sphere_type`. These two types are the *higher-level* compound types that fit better with your scenario. Once you have customized your compound types, you can use them as templates to create two instances of spheres (initialize two local variables `sphere1` and `sphere2`):
+You can use the two functions `ti.types.matrix()` and `ti.types.vector()` to create your own matrix and vector types:
+
+```python
+vec4d = ti.types.vector(4, ti.f64)  # a 64-bit floating-point 4D vector type
+mat4x3i = ti.types.matrix(4, 3, int)  # a 4x3 integer matrix type
+```
+
+You can directly call the customized vector/matrix type to instantiate vectors/matrices or use them as type hints in function arguments and struct members. For example:
+
+```python
+v = vec4d(1, 2, 3, 4)  # Create a vector instance, here v = [1.0, 2.0 3.0 4.0]
+
+@ti.func
+def length(w: vec4d):  # vec4d as type hint
+    return w.norm()
+
+@ti.kernel
+def test():
+    print(length(v))
+```
+
```

Comment:
We need one more subsection here: "Relationship between ti.Vector/Matrix and ti.types.vector/matrix"